src/sentry/models/projectownership.py

```diff
@@ -310,42 +315,55 @@ def handle_auto_assignment(
                 if not auto_assigned and not force_autoassign:
                     logger.info("autoassignment.post_manual_assignment", extra=logging_extra)
                     return
+
+            if not isinstance(owner, Team) and not isinstance(owner, RpcUser):
+                logging_extra["owner_type"] = str(type(owner))
+                logger.info("handle_auto_assignment.unknown_owner_type", extra=logging_extra)
+                return
+
             if (
                 isinstance(owner, Team)
-                and not GroupAssignee.objects.filter(group=group, team=owner.id).exists()
-            ) or (
+                and GroupAssignee.objects.filter(group=group, team=owner.id).exists()
+            ):
+                logger.info("handle_auto_assignment.team_already_assigned", extra=logging_extra)
+                return
+
+            if (
                 isinstance(owner, RpcUser)
-                and not GroupAssignee.objects.filter(group=group, user_id=owner.id).exists()
+                and GroupAssignee.objects.filter(group=group, user_id=owner.id).exists()
             ):
-                assignment = GroupAssignee.objects.assign(
-                    group,
-                    owner,
-                    create_only=not force_autoassign,
-                    extra=activity_details,
-                    force_autoassign=force_autoassign,
-                )
+                logger.info("handle_auto_assignment.user_already_assigned", extra=logging_extra)
+                return
+
+            assignment = GroupAssignee.objects.assign(
+                group,
+                owner,
+                create_only=not force_autoassign,
+                extra=activity_details,
+                force_autoassign=force_autoassign,
+            )
```

Comment:
This shouldn't be a logical change, but with multiple conditionals I'd appreciate a double check 🙏